packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -7,50 +7,76 @@
  */
 
 const fs = require('fs');
+const path = require('path');
 const crypto = require('crypto');
 const _ = require('lodash');
 const toArray = require('stream-to-array');
-const uuid = require('uuid/v4');
-
-function niceHash(buffer) {
-  return crypto
-    .createHash('sha256')
-    .update(buffer)
-    .digest('base64')
-    .replace(/=/g, '')
-    .replace(/\//g, '-')
-    .replace(/\+/, '_');
-}
+const filenamify = require('filenamify');
+
+const randomSuffix = () => crypto.randomBytes(5).toString('hex');
+const generateFileName = name => {
+  const baseName = filenamify(name, { replacement: '_' }).replace(/\s/g, '_');
+
+  return `${baseName}_${randomSuffix()}`;
+};
 
 module.exports = {
-  bufferize: async files => {
-    if (_.isEmpty(files) === 0) {
-      throw 'Missing files.';
+  formatFileInfo({ filename, type, size }, fileInfo = {}, metas = {}) {
+    const ext = path.extname(filename);
+    const baseName = path.basename(filename, ext);
+
+    const usedName = fileInfo.name || baseName;
+
+    const entity = {
+      name: usedName,
+      alternativeText: fileInfo.alternativeText,
+      caption: fileInfo.caption,
+      // sha256: niceHash(buffer),
```

Comment:
You want to keep this until we find why we added it?

Reply:
Yep :)